### PR TITLE
Update of the docs generation script

### PR DESCRIPTION
**Pull request by meln1k**
Summary: Docs generation script is updated to support custom branch override, and we don't create temp copies anymore.

JIRA issues: MARATHON-8132

### CODE REVIEW

File: ci/generate_docs.sc [hunk 1]
```diff
@@ -21,34 +19,37 @@ import $ivy.{
...
-    if (possiblyChangedLines.nonEmpty) {
-      val msg = s"Git repository isn't clean, aborting docs generation. Changed files:\n${possiblyChangedLines.mkString("\n")}"
-      println(msg)
-      exit()
-    }
+def main(release_commits_override: Map[String, String] = Map.empty) = {
 
-    val docsBuildDir = makeTmpDir()
-    val docsSourceDir = pwd/up/"docs"
+  overrideTargetVersions = release_commits_override.mapValues(Commit)
```
Comment (jeschkies) [inline ci/generate_docs.sc]: What is this used for?

File: ci/generate_docs.sc [hunk 2]
```diff
@@ -129,33 +133,43 @@ def prepareDockerImage(buildDir: Path, docsDir: Path): Unit = {
   rm ! imagePreparationWorkingDir
 }
 
-def generateTopLevelDocs(buildDir: Path, docsDir: Path) = {
+def generateTopLevelDocs(buildDir: Path, docsDir: Path, checkedRepoDir: Path) = {
```
Comment (jeschkies) [inline ci/generate_docs.sc]: Could you add a doc string? I'm not sure I know what `checkedRepoDir` ist/

File: ci/generate_docs.sc [hunk 3]
```diff
@@ -197,9 +213,28 @@ def launchPreview(buildDir: Path): Unit = {
   bindingFuture.flatMap(_.unbind()).onComplete(_ => system.terminate()) // and shutdown when done
 }
 
-def buildDocs(buildDir: Path, docsDir: Path) = {
+def copyMarathon(buildDir: Path): Path = {
+  println(s"Copying marathon folder into $buildDir")
+  cp.into(pwd/up, buildDir)
```
Comment (jeschkies) [inline ci/generate_docs.sc]: Why use this instead of a temp folder?

File: ci/generate_docs.sc [hunk 4]
```diff
@@ -77,17 +80,22 @@ def getLatestPatch(tags: Seq[SemVer]): SemVer = {
...
+    .groupBy(version => s"${version.major}.${version.minor}")
+    .filterKeys(notIgnoredBranch)
+    .mapValues(getLatestPatch)
+    .mapValues(Tag)
+    .toList
+    .sortBy(_._1)
+    .takeRight(3)
+    .map { case (releaseVersion, tag) =>
+      val tagReplacement = overrideTargetVersions.getOrElse(releaseVersion, tag)
+        releaseVersion -> tagReplacement
```
Comment (ichernetsky) [inline ci/generate_docs.sc]: Looks like bad indentation.

File: ci/generate_docs.sc (hunk 3 above)
Comment (meln1k) [reply ci/generate_docs.sc]: well, build dir is a temp folder already :)

**Pull request merged**